mypyc/emitclass.py

```diff
@@ -100,12 +100,14 @@ def generate_class(cl: ClassIR, module: str, emitter: Emitter) -> None:
     dealloc_name = '{}_dealloc'.format(name_prefix)
     methods_name = '{}_methods'.format(name_prefix)
     vtable_setup_name = '{}_trait_vtable_setup'.format(name_prefix)
+    async_struct_name = '{}_async_struct'.format(name_prefix)
```

Comment:
There is actually already support for generating slots from dunder methods in a generic table-driven way: https://github.com/mypyc/mypyc/blob/master/mypyc/emitclass.py#L31-L60

I think it should work to use that directly.

Reply:
Done!